packages/@ourworldindata/grapher/src/barCharts/DiscreteBarChart.tsx

```diff
@@ -405,11 +417,15 @@ export class DiscreteBarChart
                 {this.hasColorLegend && (
                     <HorizontalNumericColorLegend manager={this} />
                 )}
-                <HorizontalAxisComponent
-                    bounds={boundsWithoutColorLegend}
-                    axis={yAxis}
-                    preferredAxisPosition={innerBounds.bottom}
-                />
+                {/* only show the axis if all columns share a unit */}
+                {new Set(this.yColumns.map((column) => column.unit)).size ===
```

Comment:
In addition, I think we should use `shortUnit` here: It's what we display on the axis.

```suggestion
                {new Set(this.yColumns.map((column) => column.shortUnit)).size ===
```

### Reasoning
I can imagine cases where long units are verbose like `% of males` and `% of females`, and the corresponding short unit is `%` for both.
In this case, they are of course comparable.

We have a similar situation in [this chart](https://ourworldindata.org/grapher/births-and-deaths-projected-to-2100?time=earliest), where the long units are `births` and `deaths`, but the short unit is empty. This chart definitely should show an axis.